prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirBundleUtils.kt

```diff
@@ -42,7 +42,7 @@ object FhirBundleUtils : Logging {
     fun convertFhirType(value: Base, sourceType: String, targetType: String): Base {
         return if (sourceType == targetType || targetType == "*") {
             value
-        } else if (StringCompatibleType.values().any { it.typeAsString == sourceType }) {
+        } else if (sourceType == "integer" || StringCompatibleType.values().any { it.typeAsString == sourceType }) {
```

Comment:
Why is "integer" not a string compatible type? I don't like having to "manually" manage additional types here.

Reply:
I went back and forth on this, basically I wanted to support converting from an integer to a string-like type, without wanting to deal with converting from other string-like types to integer because integer is so restrictive. But the same problem exists for some of the other types like date/datetime anyway, so maybe this should be handled the same way as all of those.